jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1455,7 +1456,9 @@ Method getMethodHelper(
 	boolean throwException, boolean forDeclaredMethod, List<Method> methodList, String name, Class<?>... parameterTypes)
 	throws NoSuchMethodException {
 	Method result;
+	Method resultFromSuperClass = null;
 	Method bestCandidate;
+	Method candidateFromTopSuperClass;
```

Comment:
This variable name is not accurate since it is not just top super class. Please update another variable `declaringClassFromTopSuperClass` and  related comments as well.

Reply:
Updated as suggested.